### PR TITLE
Make networking questions clearer

### PR DESCRIPTION
**Pull request by gnuoy**
This PRs seeks to make the networking questions clearer.

### CODE REVIEW

**Comment by gnuoy** [comment]
Example output:

```
$ microstack configure 
Initializing Terraform from provider mirror ... done
Username to use for access to OpenStack (demo): 
Password to use for access to OpenStack (nAq9QmLlxtRb): 
Network range to use for project network (192.168.122.0/24): 
Setup security group rules for SSH and ICMP ingress [y/n] (y): 
Access to guests from this machine only. (Wider network access requires an additional network interface). [y/n] (y): 
CIDR of network to use for external networking. (This network is exclusively for microstacks use and not exist anywhere else on the local network) (10.20.20.0/24): 
IP address microstack will assign for traffic to/from externalnetwork. (This address should be currently unassigned.) (10.20.20.1): 
Start of IP allocation range for external network (10.20.20.2): 
End of IP allocation range for external network (10.20.20.254): 
Neutron label for physical network to map to external network (physnet1): 
Network type for access to external network [flat/vlan] (flat): 
Configuring OpenStack cloud for use ... done
# openrc for demo
export OS_AUTH_URL=http://10.177.200.172:80/openstack-keystone
export OS_USERNAME=demo
export OS_PASSWORD=nAq9QmLlxtRb
export OS_USER_DOMAIN_NAME=users
export OS_PROJECT_DOMAIN_NAME=users
export OS_PROJECT_NAME=demo
export OS_AUTH_VERSION=3
export OS_IDENTITY_API_VERSION=3
Generating openrc for cloud usage ... done
Updating hypervisor external network configuration ... done
```

File: sunbeam/commands/configure.py [hunk 1]
```diff
@@ -52,6 +52,13 @@ def user_questions():
         "security_group_rules": question_helper.ConfirmQuestion(
             "Setup security group rules for SSH and ICMP ingress", default_value=True
         ),
+        "enable_host_only_networking": question_helper.ConfirmQuestion(
+            (
+                "Access to guests from this machine only. (Wider network access "
```
Comment (pmatulis) [inline sunbeam/commands/configure.py:57]: This question can infer to users that external access is possible if it is answered in the negative: "I do *not* want access constrained only to the localhost; I want to access the VMs anywhere from the network."

The text in parentheses suggests that if the question is answered in the negative (according to the above logic) that there will be a further question asking what interface should be used for external access. Or a user may conclude that the software has analysed the host and determined that there *is no* other interface available and offered this particularly phrased question, but I don't think this is the case.

Can we assume that access from at least the machine is always desirable? If so, we can then ask directly whether external access is also wanted. If the software is not ready for that then we simply omit this line of questioning.

Comment (gnuoy) [reply sunbeam/commands/configure.py]: The questions now read:
```
$ microstack configure 
Initializing Terraform from provider mirror ... done
Username to use for access to OpenStack (demo): 
Password to use for access to OpenStack (X7VpuIvK6Jw6): 
Network range to use for project network (192.168.122.0/24): 
Setup security group rules for SSH and ICMP ingress [y/n] (y): 
Local or remote access to VMs [local/remote] (local): 
CIDR of network to use for external networking. (This network is exclusively for microstacks use and not exist anywhere else on the local network) (10.20.20.0/24): 
IP address microstack will assign for traffic to/from externalnetwork. (This address should be currently unassigned.) (10.20.20.1): 
Start of IP allocation range for external network (10.20.20.2): 
End of IP allocation range for external network (10.20.20.254): 
Neutron label for physical network to map to external network (physnet1): 
Network type for access to external network [flat/vlan] (flat): 
Configuring OpenStack cloud for use ... done
# openrc for demo
export OS_AUTH_URL=http://10.177.200.170:80/openstack-keystone
export OS_USERNAME=demo
export OS_PASSWORD=X7VpuIvK6Jw6
export OS_USER_DOMAIN_NAME=users
export OS_PROJECT_DOMAIN_NAME=users
export OS_PROJECT_NAME=demo
export OS_AUTH_VERSION=3
export OS_IDENTITY_API_VERSION=3
Generating openrc for cloud usage ... done
Updating hypervisor external network configuration ... done
```

Comment (gnuoy) [reply sunbeam/commands/configure.py]: I forgot to say I think the idea of adding a `Unused network interface on this host for remote access [e.g. /dev/enp5s0]:` question is a great one but the work needed for that is beyond the scope of this PR. However, adding that question should happen in the next two weeks

**Comment by pmatulis** [comment]
I've left suggestions for questions (to eventually be buttressed by documentation).

> `CIDR of network to use for external networking. (This network is exclusively for microstacks use and not exist anywhere else on the local network) (10.20.20.0/24):`

Suggestion:

`CIDR of OpenStack external network - arbitrary but must not be in use (10.20.20.0/24):`

I'm using "OpenStack" here to clarify that this external network has nothing to do with an actual external network; it's an OpenStack thing. Also, "local" (previous question) and "external" can sound contradictory.

Did we say we could have the software probe as to whether or not the external network is currently in use?

---

> `IP address microstack will assign for traffic to/from externalnetwork. (This address should be currently unassigned.) (10.20.20.1): 

Suggestion:

`IP address of router on the external network (10.20.20.1):`

---

> `Neutron label for physical network to map to external network (physnet1):`

Do we need to expose this?

---

> `Local or remote access to VMs [local/remote] (local):`

What ensues if remote is chosen?

**Comment by pmatulis** [comment]
I'm now wondering whether we need to prompt for the router address.

**Comment by gnuoy** [comment]
Questions for local install now look like:

```
Initializing Terraform from provider mirror ... done
Username to use for access to OpenStack (demo): 
Password to use for access to OpenStack (McKdMEg8Ufm9): 
Network range to use for project network (192.168.122.0/24): 
Setup security group rules for SSH and ICMP ingress [y/n] (y): 
Local or remote access to VMs [local/remote] (local): 
CIDR of OpenStack external network - arbitrary but must not be in use (10.20.20.0/24): 
Start of IP allocation range for external network (10.20.20.2): 
End of IP allocation range for external network (10.20.20.254): 
Network type for access to external network [flat/vlan] (flat): 
Configuring OpenStack cloud for use ... done
# openrc for demo
export OS_AUTH_URL=http://10.177.200.170:80/openstack-keystone
export OS_USERNAME=demo
export OS_PASSWORD=McKdMEg8Ufm9
export OS_USER_DOMAIN_NAME=users
export OS_PROJECT_DOMAIN_NAME=users
export OS_PROJECT_NAME=demo
export OS_AUTH_VERSION=3
export OS_IDENTITY_API_VERSION=3
Generating openrc for cloud usage ... done
Updating hypervisor external network configuration ... done
```

**Comment by gnuoy** [comment]
> I've left suggestions for questions (to eventually be buttressed by documentation).
> 
> > `CIDR of network to use for external networking. (This network is exclusively for microstacks use and not exist anywhere else on the local network) (10.20.20.0/24):`
> 
> Suggestion:
> 
> `CIDR of OpenStack external network - arbitrary but must not be in use (10.20.20.0/24):`
> 
> I'm using "OpenStack" here to clarify that this external network has nothing to do with an actual external network; it's an OpenStack thing. Also, "local" (previous question) and "external" can sound contradictory.
> 
> Did we say we could have the software probe as to whether or not the external network is currently in use?
> 
> > `IP address microstack will assign for traffic to/from externalnetwork. (This address should be currently unassigned.) (10.20.20.1):
> 
> Suggestion:
> 
> `IP address of router on the external network (10.20.20.1):`
> 
> > `Neutron label for physical network to map to external network (physnet1):`
> 
> Do we need to expose this?
> 
> > `Local or remote access to VMs [local/remote] (local):`
> 
> What ensues if remote is chosen?

If remote is chosen you can access your guests from other machines on the network but not from the local machine. If local is chosen you can access your guests from the local machine but not from other machines on the network

**Comment by gnuoy** [comment]
> > I've left suggestions for questions (to eventually be buttressed by documentation).
> > > `CIDR of network to use for external networking. (This network is exclusively for microstacks use and not exist anywhere else on the local network) (10.20.20.0/24):`
> > 
> > 
> > Suggestion:
> > `CIDR of OpenStack external network - arbitrary but must not be in use (10.20.20.0/24):`
> > I'm using "OpenStack" here to clarify that this external network has nothing to do with an actual external network; it's an OpenStack thing. Also, "local" (previous question) and "external" can sound contradictory.
> > Did we say we could have the software probe as to whether or not the external network is currently in use?
> > > `IP address microstack will assign for traffic to/from externalnetwork. (This address should be currently unassigned.) (10.20.20.1):
> > 
> > 
> > Suggestion:
> > `IP address of router on the external network (10.20.20.1):`
> > > `Neutron label for physical network to map to external network (physnet1):`
> > 
> > 
> > Do we need to expose this?
> > > `Local or remote access to VMs [local/remote] (local):`
> > 
> > 
> > What ensues if remote is chosen?
> 
> If remote is chosen you can access your guests from other machines on the network but not from the local machine. If local is chosen you can access your guests from the local machine but not from other machines on the network

Sorry, I misunderstood your question. If `remote` is selected the questions are:
```
Initializing Terraform from provider mirror ... done
Username to use for access to OpenStack (demo): 
Password to use for access to OpenStack (737DzptNrYfs): 
Network range to use for project network (192.168.122.0/24): 
Setup security group rules for SSH and ICMP ingress [y/n] (y): 
Local or remote access to VMs [local/remote] (local): remote
CIDR of network to use for external networking (10.20.20.0/24): 
IP address of gateway for external network (10.20.20.1): 
Start of IP allocation range for external network (10.20.20.2): 
End of IP allocation range for external network (10.20.20.254): 
Network type for access to external network [flat/vlan] (flat): 
Configuring OpenStack cloud for use ... done
# openrc for demo
export OS_AUTH_URL=http://10.177.200.170:80/openstack-keystone
export OS_USERNAME=demo
export OS_PASSWORD=737DzptNrYfs
export OS_USER_DOMAIN_NAME=users
export OS_PROJECT_DOMAIN_NAME=users
export OS_PROJECT_NAME=demo
export OS_AUTH_VERSION=3
export OS_IDENTITY_API_VERSION=3
Generating openrc for cloud usage ... done
Updating hypervisor external network configuration ... done
```

**Comment by pmatulis** [comment]
I guess we should make the analogous changes/removals for these right?

```
CIDR of network to use for external networking (10.20.20.0/24): 
IP address of gateway for external network (10.20.20.1):
```

**Comment by gnuoy** [comment]
> I guess we should make the analogous changes/removals for these right?
> 
> ```
> CIDR of network to use for external networking (10.20.20.0/24): 
> IP address of gateway for external network (10.20.20.1):
> ```

I think they should stay as they are as they don't refer to something the snap is creating but to the users exiting network and gateway.

**Comment by pmatulis** [comment]
Sounds good. Just `default gateway` :wink:

**Comment by gnuoy** [comment]
The sunbeam/edge/net-questions branch of the microstack snap contains this change

**Review by javacruft — APPROVED** [review]

**Pull request merged**